src/app/viewer/page.tsx

```diff
@@ -0,0 +1,262 @@
+    setNameError(false);
+    setPasscodeError(false);
+
+    // Validate the form
+    if (!name) {
+      setNameError(true);
+      return;
+    }
+
+    // Validate and sanitize the URL
+    const url = tokenData?.url;
+    if (!url) {
+      setError(
+        'Invalid token data. Please ask the patient to generate a new valid link.',
+      );
+      return;
+    } else if (!isValidUrl(url)) {
+      setError('Invalid URL detected. Please contact support.');
+      return;
+    }
+    if (tokenData && tokenData.flag !== 'L' && !passcode) {
+      setPasscodeError(true);
+      return;
+    }
+
+    // Extract URL from tokenData
+    if (!url) {
+      setError(
+        'Invalid token data. Please ask the patient to generate a new valid link.',
+      );
+      return;
+    }
+
+    // Prepare the request body
+    const requestBody: { recipient: string; passcode?: string } = {
+      recipient: name,
+    };
+
+    if (tokenData?.flag !== 'L') {
+      requestBody.passcode = passcode;
+    }
+
+    try {
+      setLoading(true);
+      // Send POST request using fetchShareLinkData
+      const response = await apiViewer.fetchShareLinkData(url, requestBody);
+      if (response.status !== 200) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+      }
+
+      const responseData = response.data;
+
+      // Check if files array is available and extract location
+      if (responseData.files && responseData.files.length > 0) {
+        const location = responseData.files[0].location;
+        // Run GET request using the value of location as URL
+        const fileResponse = await apiViewer.getShareLinkData(location);
+        setFhirBundle(fileResponse.data);
+      } else if (responseData.files && responseData.files.length === 0) {
+        setError(
+          'The SHlink has no files associated with it. Please ask the patient to generate a new valid link.',
+        );
+      }
+    } catch (error) {
+      setError(
+        `Error submitting form: ${error instanceof Error ? error.message : String(error)}`,
+      );
+      if (axios.isAxiosError(error)) {
+        // Handle Axios-specific errors
+        if (error.response) {
+          // Server responded with a status other than 200 range
+          console.log(error.response);
+          setError(
+            `Error: ${error.response.status} - ${JSON.stringify(error.response.data)}`,
+          );
+        } else if (error.request) {
+          // Request was made but no response was received
+          setError('Error: No response received from the server');
+        } else {
+          // Something happened in setting up the request
+          setError(`Error: ${error.message}`);
+        }
+      } else {
+        // Handle other types of errors
+        setError('Error submitting form');
+      }
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  if (loading) {
+    return <PatientSummarySkeleton />;
+  }
+
+  if (!isAuthenticated) {
```

Comment:
A user should not need to be logged in to view the link

Reply:
I guess I pushed the changes at the same time you were reviewing the PR. This was addressed in  a2d5604d9133f6d9e4a96327298d64b4de345a4b